CommonTools/MVAUtils/src/GBRForestTools.cc

```diff
@@ -0,0 +1,302 @@
+      if (nodeIsTerminal) {
+        double response = 0.;
+        if (isRegression) {
+          node->QueryDoubleAttribute("res", &response);
+        }
+        else {
+          if (useYesNoLeaf) {
+            node->QueryDoubleAttribute("nType", &response);
+          }
+          else {
+            if (isAdaClassifier) {
+                node->QueryDoubleAttribute("purity", &response);
+            } else {
+                node->QueryDoubleAttribute("res", &response);
+            }
+          }
+        }
+        response *= scale;
+        tree.Responses().push_back(response);
+      }
+      else {    
+
+        int thisidx = tree.CutIndices().size(); 
+        
+        int selector; 
+        float cutval; 
+        bool ctype; 
+
+        node->QueryIntAttribute("IVar", &selector);
+        node->QueryFloatAttribute("Cut", &cutval);
+        node->QueryBoolAttribute("cType", &ctype);
+
+        tree.CutIndices().push_back(static_cast<unsigned char>(selector)); 
+
+        //newer tmva versions use >= instead of > in decision tree splits, so adjust cut value
+        //to reproduce the correct behaviour
+        if (adjustboundary) {
+          cutval = std::nextafter(cutval,std::numeric_limits<float>::lowest());
+        }
+        tree.CutVals().push_back(cutval);
+        tree.LeftIndices().push_back(0);   
+        tree.RightIndices().push_back(0);
+        
+        tinyxml2::XMLElement* left = nullptr;
+        tinyxml2::XMLElement* right = nullptr;
+        for(tinyxml2::XMLElement* e = node->FirstChildElement("Node");
+                e != nullptr; e = e->NextSiblingElement("Node")) {
+            if (*(e->Attribute("pos")) == 'l') left = e;
+            else if (*(e->Attribute("pos")) == 'r') right = e;
+        }
+        if (!ctype) {
+          std::swap(left, right);
+        }
+
+        tree.LeftIndices()[thisidx] = isTerminal(left) ? -tree.Responses().size() : tree.CutIndices().size() ;
+        addNode(tree, left, scale, isRegression, useYesNoLeaf, adjustboundary,isAdaClassifier);
+        
+        tree.RightIndices()[thisidx] = isTerminal(right) ? -tree.Responses().size() : tree.CutIndices().size() ;
+        addNode(tree, right, scale, isRegression, useYesNoLeaf, adjustboundary,isAdaClassifier);
+        
+      }
+  
+    }
+
+    std::unique_ptr<GBRForest> init(const std::string& weightsFileFullPath,
+                                    std::vector<std::string>& varNames)
+    {
+
+      std::string method;
+      //
+      // Load weights file, for gzipped or raw xml file
+      //
+      tinyxml2::XMLDocument xmlDoc;
+
+      if (reco::details::hasEnding(weightsFileFullPath, ".xml")) {
+          xmlDoc.LoadFile(weightsFileFullPath.c_str());
+      } else if (reco::details::hasEnding(weightsFileFullPath, ".gz") ||
+                 reco::details::hasEnding(weightsFileFullPath, ".gzip")) {
+          xmlDoc.Parse(reco::details::readGzipFile(weightsFileFullPath));
```

Comment:
```reco::details::readGzipFile``` creates but doesn't own the output buffer. So, it's leaking out here.
So, this line here adds about 200MB, which is a clear showstopper.
Please cleanup appropriately.